### ISSUE TITLE
Declutter Android Auto

### ISSUE THREAD

**Issue by GSzabados** [home-assistant/android#3219]
**Is your feature request related to a problem? Please describe.**

As #791 has been closed and Android Auto support is available. It would be nice to declutter it from possibly dozens or hundreds of entities.

**Describe the solution you'd like**

One option to solve the problem is to add a Favourites option where only selected entities will show up. Like what is needed from the car. 

**Describe alternatives you've considered, if any**

The other option is to provide an option from the app where only selected entities would get into the Android Auto display under each category. 

**Additional context**

In my case, I have only 1 real door lock, but all the Child Locks from Z2M shows up under the locks section. Selecting only the single lock which is needed in Android Auto would make it less cluttered.

I believe everyone who has an extensive setup of device will see the same issue as all switches, lights, etc shows up and some might not needed when driving and just distracts meanwhile searching for the right one.

**Comment by dshokouhi** [comment]
I like the favorites suggestion as it falls in line with Wear OS where if a user select favorite entities they appear first and then just below all the other entities will be loaded.

**Comment by JBassett** [comment]
It's on my list of things to tackle.

**Comment by GSzabados** [comment]
Just to add as some additional information. 

The favourite way is good for easy access for selected entities, but it seems to be that the additional selection for entities would do a really good job as well.

As I wrote before, all the child locks for my Zigbee sockets ends up under the the lock category, and makes it really messy.
Also, I had to realise that from Zwave all the Ping push buttons ends up in the buttons category, and makes that as well messy.
The ESPHome switches which used for restarting ESPHome devices end up in the switch category as well.

Unfortunately some of these fall under Configuration but some are Controls in HA's device/entity categories.
So no easy solution just to disregard Configuration entities. 

Also, groups are not listed, neither if they are lights or switches. But I think that is different issue.

**Comment by dshokouhi** [comment]
> Also, groups are not listed, neither if they are lights or switches. But I think that is different issue.

Yes `group` is not part of the initial supported domains

https://github.com/home-assistant/android/blob/master/app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt#L51-L61

you can open a separate feature request for domains not currently shown :)

**Comment by GSzabados** [comment]
> you can open a separate feature request for domains not currently shown :)

I will open one definitely.

I have to open a bug also, when I will have some time. The navigation in the menu is quite clunky. I need to talk with some other AA users first, because I cannot be sure that is it my head unit only or the way how it is working.

**Comment by simone7121** [comment]
Or another thing on how they can implement, is the use of an home screen with the principal entities, like the SmartThings app or better with a simil Lovelace UI. It would be a nice idea though. Because it's difficult to see an navigate through all the menus and entities.

**Comment by KidA001** [comment]
Want to add a +1 for displaying a list of favorite entities or having a custom view/home screen for Android Auto to make navigation simpler. I echo a lot of the sentiments already shared here. Really excited to have this feature.

@GSzabados I am happy to do some testing on my AA head unit. Let me know if there are some steps I can do to try and reproduce.

**Comment by ArekKubacki** [comment]
Or maybe like in garmin watches, where entity groups and scenes are given. This seems to be the optimal solution. Showing all entities does not make much sense with an extensive system.

**Comment by andynash** [comment]
+1 for a user selected group of favourites that are likely to be useful from the car 👍 A great second iteration to improve the UX.

+2 if that group can be automated in some way (perhaps in a later release of course!), for example when near home show a certain group of entities/automations (I understand there is a single click requirement from Google?) and when further away how a different selection.

The latter would be a little more 'Home Assistanty' - especially in comparison to the rather rigid SmartThings implementation with a fixed set of favourites.

However the dependency need not be location - presumably so long as the entities/automations etc qualify for inclusion (by domain for example) then it doesn't matter how the HA user determines this list in the background, so any conditions could be permitted to influence the selection?

Other conditions could be electric/liquid fuel level, time of day and many others.

**Comment by teamllamauk** [comment]
I was wondering if it would be possible to have a specific android auto dashboard that you could setup in HA and have that as the top link when using in android auto. You could then have all of the devices you needs from the car listed in there regardless of their types. 

For instance, in an ideal world, i would like to be able to switch on certain lights but some are listed as switches and others as lights. It would be nice to group them in one place.

**Comment by qupada** [comment]
Also willing to help with testing.

I've linked my garage door opener to HA via my LoRa gateway (using a relay-switched GPIO).  Waiting on some parts to be delivered to be able to fully set it up as an MQTT "Cover", so for now it's just a button.

Unfortunately this puts it in a giant list with - I just counted it - 42 other buttons, belonging to all manner of devices.  But hey, if I want to trigger the cleaning cycle of my Litter Robot (cat toilet) from the car, I'm covered for that too.

I'm of the opinion that the dashboard for the car should contain **nothing** by default, everything you put on it should be for a reason.  For instance, I'd only ever want my garage door, garage and outside lights, HVAC, and some macros to turn off all the lights in the house.  Even with the stunningly low-resolution display on my AA headunit, should just  fit on a single screen (no scrolling required would be ideal for eyes-on-road safety).

**Comment by andynash** [comment]
Agreed on the 'empty by default' suggestion to keep things simple (and safe).

@dshokouhi I'm not familiar with Wear - when you say a similar favourites list would this essentially provide a menu option at the top in Drive mode containing user-selected entities?

This would I guess be a good alternative if so, and I guess if you _really_ need something else, and the current menu items for all the different entity types was still there, you have the option of going to look for it.

**Comment by dshokouhi** [comment]
> @dshokouhi I'm not familiar with Wear - when you say a similar favourites list would this essentially provide a menu option at the top in Drive mode containing user-selected entities?

On Wear OS favorite entities are listed first before any areas or entity domains are listed. They are first to load when the app loads.

The implementation in Android Auto is still up in the air as nobody has started working on it. Its quite involved as we need to account for multi server support now which the Wear OS screen does not do.

**Comment by GSzabados** [comment]
@jpelgrom, I am really looking forward to try this fix #3670! Sounds awesome! And many thanks for the update!

**Comment by qupada** [comment]
Signed beta release (side-loadable over the play store installed version) with this feature is available, for anyone itching to give it a whirl:

https://github.com/home-assistant/android/releases/tag/2023.7.7

**Comment by SiboutVanLoo** [comment]
When is the official release planned?
Not shure if i'd wait or go ahead and scratch the itch! 🤣

**Comment by GSzabados** [comment]
It should be already part of the 2023.7.8-full release. But you can join the Beta anytime and don't need to sideload anything.

**Comment by dshokouhi** [comment]
> It should be already part of the 2023.7.8-full release. But you can join the Beta anytime and don't need to sideload anything. 

That's still a beta release. We have not cut a production release yet. We have a bug to fix before we go live

https://github.com/home-assistant/android/issues/3762

**Comment by dshokouhi** [comment]
The production release is now live for those who have not joined the beta. Please update to 2023.8.2-full and setup your favorite entities!
